bin/functions/functions.R

```diff
@@ -1,8 +1,17 @@
 if(!require("splines")) install.packages("splines")
 
 # Parse dose response data -----
-# Function to check input data format
-data_checks <- function(dat, dr_threshold) {
+
+# Function to read in data
+parse_data <- function(in_val, in_type, dr_threshold = 0.8) {
```

Comment:
I'm a bit unsure about the dr_threshold of 0.8. So, there could be a situation where if you have, say 10 doses, and all but the last dose have 3 reps, but the last one has 1 rep, the code will keep going. But it doesn't appear that last dose is removed. I don't have a problem with, per se, but I think this needs to be made more clear in the documentation.

It appears 5 doses is the smallest number of doses where we could have 1 dose that doesn't meet the 3 rep reequirement. At that point, ceiling(5 * 0.8) = 4, so that's the first point where we could have a situation where 4 doses have 3 reps and one doesn't. I don't have any issues with us accepting studies with fewer reps that meet these stipulations -- b/c that's still requiring 4 doses to meet the minimum requirements.

If you could update the documentation to make that point clear, I think we'll be good.

Reply:
I would prefer to remove it completely. One of the benefits of doing the bootstrap is that we can understand some of the variability in the response. Having any dose with an insufficient number of replicates would just tether the curve to one location which might result in a tighter 95% credible interval since we're giving that one response more weight than it's worth.

Reply:
Concur. Works for me.

Reply:
Updated at line 36 of functions.R. 

It now removes doses with fewer than 3 replicates and then checks for the minimum four doses. Updated documentation.